src/lib/tls/tls13/tls_extensions_key_share.cpp

```diff
@@ -139,21 +161,22 @@ class Key_Share_ServerHello {
 
       bool empty() const { return m_server_share.empty(); }
 
-      const Key_Share_Entry& get_singleton_entry() const { return m_server_share; }
+      Key_Share_Entry& get_singleton_entry() { return m_server_share; }
 
-      secure_vector<uint8_t> exchange(const Key_Share_ClientHello& client_shares,
-                                      const Policy& policy,
-                                      Callbacks& cb,
-                                      RandomNumberGenerator& rng) const;
+      const Key_Share_Entry& get_singleton_entry() const { return m_server_share; }
 
       std::vector<Named_Group> offered_groups() const { return {selected_group()}; }
 
       Named_Group selected_group() const { return m_server_share.group(); }
 
-      void erase() { m_server_share.erase(); }
+      secure_vector<uint8_t> take_shared_secret() {
+         BOTAN_STATE_CHECK(!m_shared_secret.empty());
+         return std::exchange(m_shared_secret, {});
```

Comment:
It seems clang 11 does not know about `std::exchange` :disappointed:

Reply:
We're using it in other places, though. I rather think, an explicit `#include <utility>` should do the trick.